collectors/mesos/agent/container.go

```diff
@@ -0,0 +1,88 @@
+// Copyright 2016 Mesosphere, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package agent
+
+import (
+	"net"
+	"net/url"
+	"strconv"
+
+	"github.com/dcos/dcos-metrics/util/http/client"
+)
+
+// agentContainer defines the structure of the response expected from Mesos
+// *for a single container* when polling the '/containers' endpoint in API v1.
+// Note that agentContainer is actually in a top-level array. For more info, see
+// https://github.com/apache/mesos/blob/1.0.1/include/mesos/v1/agent/agent.proto#L161-L168
+type agentContainer struct {
+	FrameworkID     string                 `json:"framework_id"`
+	ExecutorID      string                 `json:"executor_id"`
+	ExecutorName    string                 `json:"executor_name"`
+	Source          string                 `json:"source"`
+	ContainerID     string                 `json:"container_id"`
+	ContainerStatus map[string]interface{} `json:"container_status"`
+	Statistics      *resourceStatistics    `json:"statistics"`
+}
+
+// resourceStatistics defines the structure of the response expected from Mesos
+// when referring to container and/or executor metrics. In Mesos, the
+// ResourceStatistics message is very large; it defines many fields that are
+// dependent on a feature being enabled in Mesos, and not all of those features
+// are enabled in DC/OS.
+//
+// Therefore, we redefine the resourceStatistics struct here with only the fields
+// dcos-metrics currently cares about, which should be stable for Mesos API v1.
+//
+// For a complete reference, see:
+// https://github.com/apache/mesos/blob/1.0.1/include/mesos/v1/mesos.proto#L921-L1022
+type resourceStatistics struct {
```

Comment:
that's an idea, I hadn't really thought much about using this project as a library. If we did,  I'd want to push it into dcos-go (maybe make that dcos/cluster package I keep wanting) to do it?